mvrx/src/main/kotlin/com/airbnb/mvrx/ScriptableMvRxStateStore.kt

```diff
@@ -0,0 +1,34 @@
+package com.airbnb.mvrx
+
+import io.reactivex.Observable
+import io.reactivex.subjects.BehaviorSubject
+
+/**
+ * A [MvRxStateStore] which ignores standard calls to [set]. Instead it can be scripted via calls to
+ * [next]. This is intended to be used for tests only, and in particular UI tests where you wish to test
+ * how your UI code reacts to different ViewModel states. This is not as useful for unit testing your view model,
+ * as business logic in state reducers will not be used.
+ */
+class ScriptableMvRxStateStore<S : Any>(initialState: S) : MvRxStateStore<S> {
+
+    private val subject: BehaviorSubject<S> = BehaviorSubject.createDefault(initialState)
+
+    override val observable: Observable<S> = subject.distinctUntilChanged()
+
+    override val state: S
+        get() = subject.value!!
+
+    override fun get(block: (S) -> Unit) {
+        block(state)
+    }
+
+    override fun set(stateReducer: S.() -> S) {
+        // No-op set the state via next
+    }
+
+    fun next(state: S) = subject.onNext(state)
```

Comment:
set on the real store is asynchronous, we need this to be synchronous for tests. Also,  we want `set` to be a no-op so that observables in the real viewmodel don't override our test state

Reply:
i think all tests run synchronously

Reply:
@hellohuanlin This is for use in integration tests, specifically screenshot testing. Unit tests only run synchronously because we override RxJava schedulers via plugins, which are global settings, and not something we would want to change in an integration test as it would effect non-mvrx code.